tests.foreman.api.test_repository.RepositoryTestCase.test_positive_update_gpg : TypeError: 'NoneType' object is not iterable

**tests.foreman.api.test_repository.RepositoryTestCase.test_positive_update_gpg (from tier2-parallel) fails with below:**
++++++++++++++++++++++++++
self = <tests.foreman.api.test_repository.RepositoryTestCase testMethod=test_positive_update_gpg>

    @tier2
    @run_only_on('sat')
    def test_positive_update_gpg(self):
        """Create a repository and update its GPGKey
    
        :id: 0e9319dc-c922-4ecf-9f83-d221cfdf54c2
    
        :expectedresults: The updated repository points to a new GPG key.
    
        :CaseLevel: Integration
        """
        # Create a repo and make it point to a GPG key.
        gpg_key_1 = entities.GPGKey(
            content=read_data_file(VALID_GPG_KEY_FILE),
            organization=self.org,
        ).create()
        repo = entities.Repository(
            gpg_key=gpg_key_1,
            product=self.product,
        ).create()
    
        # Update the repo and make it point to a new GPG key.
        gpg_key_2 = entities.GPGKey(
            content=read_data_file(VALID_GPG_KEY_BETA_FILE),
            organization=self.org,
        ).create()
        repo.gpg_key = gpg_key_2
>       repo = repo.update()

tests/foreman/api/test_repository.py:738: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../shiningpanda/jobs/e29ba9b0/virtualenvs/d41d8cd9/lib/python3.6/site-packages/blinker_herald/base.py:137: in wrapper
    result = fn(*args, **kwargs)
../../shiningpanda/jobs/e29ba9b0/virtualenvs/d41d8cd9/lib/python3.6/site-packages/nailgun/entity_mixins.py:1061: in update
    return self.read(attrs=self.update_json(fields))
../../shiningpanda/jobs/e29ba9b0/virtualenvs/d41d8cd9/lib/python3.6/site-packages/nailgun/entity_mixins.py:1036: in update_json
    response = self.update_raw(fields)
../../shiningpanda/jobs/e29ba9b0/virtualenvs/d41d8cd9/lib/python3.6/site-packages/nailgun/entity_mixins.py:1018: in update_raw
    self.update_payload(fields),
../../shiningpanda/jobs/e29ba9b0/virtualenvs/d41d8cd9/lib/python3.6/site-packages/nailgun/entity_mixins.py:1003: in update_payload
    return _payload(self.get_fields(), values)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

fields = {'backend_identifier': <nailgun.entity_fields.StringField object at 0x7f4f38775be0>, 'checksum_type': <nailgun.entity_...ringField object at 0x7f4f38749240>, 'content_counts': <nailgun.entity_fields.DictField object at 0x7f4f38775c18>, ...}
values = {'backend_identifier': '5d0e69ba-950e-47f7-b601-99f0cf96bb1d', 'checksum_type': None, 'container_repository_name': None, 'content_counts': {'deb': 0, 'docker_manifest': 0, 'docker_manifest_list': 0, 'docker_tag': 0, ...}, ...}

    def _payload(fields, values):
        """Implement the ``*_payload`` methods.
    
        It's frequently useful to create a dict of values that can be encoded to
        JSON and sent to the server. Unfortunately, there are mismatches between
        the field names used by NailGun and the field names the server expects.
        This method provides a default translation that works in many cases. For
        example:
    
        >>> from nailgun.entities import Product
        >>> product = Product(name='foo', organization=1)
        >>> set(product.get_fields())
        {
            'description',
            'gpg_key',
            'id',
            'label',
            'name',
            'organization',
            'sync_plan',
        }
        >>> set(product.get_values())
        {'name', 'organization'}
        >>> product.create_payload()
        {'organization_id': 1, 'name': 'foo'}
    
        :param fields: A value like what is returned by
            :meth:`nailgun.entity_mixins.Entity.get_fields`.
        :param values: A value like what is returned by
            :meth:`nailgun.entity_mixins.Entity.get_values`.
        :returns: A dict mapping field names to field values.
    
        """
        for field_name, field in fields.items():
            if field_name in values:
                if isinstance(field, OneToOneField):
                    values[field_name + '_id'] = (
                        getattr(values.pop(field_name), 'id', None)
                    )
                elif isinstance(field, OneToManyField):
                    values[field_name + '_ids'] = [
                        entity.id for entity in values.pop(field_name)
                    ]
                elif isinstance(field, ListField):
                    def parse(obj):
                        """parse obj payload if it is an Entity"""
                        if isinstance(obj, Entity):
                            return _payload(obj.get_fields(), obj.get_values())
                        return obj
    
                    values[field_name] = [
>                       parse(obj) for obj in values[field_name]]
E                   TypeError: 'NoneType' object is not iterable

../../shiningpanda/jobs/e29ba9b0/virtualenvs/d41d8cd9/lib/python3.6/site-packages/nailgun/entity_mixins.py:208: TypeError

++++++++++++++++++++++++++

--- a/tests/foreman/api/test_repository.py
+++ b/tests/foreman/api/test_repository.py
@@ -735,7 +735,7 @@ class RepositoryTestCase(APITestCase):
             organization=self.org,
         ).create()
         repo.gpg_key = gpg_key_2
-        repo = repo.update()
+        repo = repo.update(['gpg_key'])
         self.assertEqual(repo.gpg_key.id, gpg_key_2.id)
 
     @run_only_on('sat')

